lib/components/regions_map/regions_map.dart

```diff
@@ -109,25 +110,41 @@ class _RegionsMapState extends State<RegionsMap> {
   }
 }
 
-class RegionsSearchBar extends StatelessWidget {
+class RegionsSearchBar extends StatefulWidget {
   const RegionsSearchBar({Key? key}) : super(key: key);
 
+  @override
+  State<RegionsSearchBar> createState() => _RegionsSearchBarState();
+}
+
+class _RegionsSearchBarState extends State<RegionsSearchBar> {
+  bool showSearch = false;
```

Comment:
can this showSearch be store in the pageState? and not in the UI?

Reply:
Mmmmmmmmmmmmmmmmmm
First at all I don't even know if this is the final UI I have to ask Jason since it doesn't have a drop down design in figma.
Second I know we put all business login in blocs but IMO I don't see bad use setState for small UI things like tooggle a bool. That save me a event a bloc mapper and a builder. If we go for consistent there are many onPress that launch a dialod /navigation and that should be done by a bloc event and handled by a listener. I'll do the changes once Jason give me the design.

Reply:
fixed

Reply:
if the page is super simple and doesnt yet have a bloc with state and all our architechture, then yeah, its cool to have this simple boolean here.

Reply:
@gguijarro-c-chwy 
Does this pull request need anything else? 
I already added the Jason's recommendations in UI.